zerver/views/streams.py

```diff
@@ -332,7 +332,7 @@ def add_subscriptions_backend(
         if user_profile.realm.is_zephyr_mirror_realm and not all(stream.invite_only for stream in streams):
             return json_error(_("You can only invite other Zephyr mirroring users to private streams."))
         if not user_profile.can_subscribe_other_users():
-            if user_profile.realm.invite_to_stream_by_admins_only:
+            if user_profile.can_only_administrators_subscribe_other_users():
```

Comment:
This change (as well as the tests) probably should just be squashed into the original commit.  There's no reason to separate test changes from main code changes.

Reply:
Fixed, thanks!